backend/eventloop/shared_resource_loop/sharedresourceloop.go

```diff
@@ -334,23 +394,585 @@ func processSharedResourceMessage(ctx context.Context, msg sharedResourceLoopMes
+
+		} else {
+			l.Info("No update needed")
+			return &dbRepoCred, nil
+		}
+	}
+}
+
+func cleanRepoCredOperation(ctx context.Context, dbRepoCred db.RepositoryCredentials, clusterUser *db.ClusterUser, ns string,
+	dbQueries db.DatabaseQueries, client client.Client, operationDBID string, l logr.Logger) error {
+
+	l.Info("Checking if there is an operation for this RepositoryCredential ready to be cleaned up")
+
+	// Get list of Operations from cluster.
+	listOfK8sOperation := managedgitopsv1alpha1.OperationList{}
+	err := client.List(ctx, &listOfK8sOperation)
+	if err != nil {
+		l.Error(err, "Unable to fetch list of k8s Operation from cluster.")
+		return err
+	}
+
+	for _, k8sOperation := range listOfK8sOperation.Items {
+		// Skip if Operation CR is not related to the Database Operation row.
+
+		// Skip if Operation is not for this RepositoryCredential.
+		if k8sOperation.Spec.OperationID != operationDBID {
+			errV := fmt.Errorf("skipping Operation (ID: %v) because it is not for this RepositoryCredential Operation Row (ID: %v)", k8sOperation.Spec.OperationID, operationDBID)
+			l.Info(errV.Error())
+			continue
+		}
+
+		// Fetch corresponding DB entry.
+		dbOperation := db.Operation{
+			Operation_id: k8sOperation.Spec.OperationID,
+		}
+
+		// If we cannot fetch the DB entry, then skip.
+		if err := dbQueries.GetOperationById(ctx, &dbOperation); err != nil {
+			l.Error(err, "Unable to fetch Operation from DB.")
+			continue
+		}
+
+		// If Operation is not in a terminal state, then skip.
+		if dbOperation.State != db.OperationState_Completed && dbOperation.State != db.OperationState_Failed {
+			l.V(sharedutil.LogLevel_Debug).Info("K8s Operation is not ready for cleanup : " + string(k8sOperation.UID) + " DbOperation: " + k8sOperation.Spec.OperationID)
+			continue
+		}
+
+		// If Operation is not for the Operation_owner_user_id of the RepositoryCredential, then skip.
+		if dbOperation.Operation_owner_user_id != clusterUser.Clusteruser_id {
+			l.Error(err, "Skipping Operation that is not for this RepositoryCredential's Operation_owner_user_id.")
+			continue
+		}
+
+		l.Info("Deleting Operation CR: " + string(k8sOperation.UID) + " (UID) and the related Operation DB entry: " + k8sOperation.Spec.OperationID + " ID")
+
+		// Delete the k8s operation now.
+		if err := operations.CleanupOperation(ctx, dbOperation, k8sOperation, ns, dbQueries, client, l); err != nil {
+			l.Error(err, "Unable to Delete k8s Operation"+string(k8sOperation.UID)+" for DbOperation: "+k8sOperation.Spec.OperationID)
+		} else {
+			l.Info("Deleted k8s Operation: " + string(k8sOperation.UID) + " for DbOperation: " + k8sOperation.Spec.OperationID)
+		}
+	}
+
+	return nil
+}
+
+func createRepoCredOperation(ctx context.Context, dbRepoCred db.RepositoryCredentials, clusterUser *db.ClusterUser, ns string,
+	dbQueries db.DatabaseQueries, apiNamespaceClient client.Client, shouldWait bool, l logr.Logger) (error, string) {
+
+	l.Info("Creating operation...")
```

Comment:
If you want leave this Info statement in, you should provide more context in the log statement message